js/ad_manager_vast.js

```diff
@@ -972,13 +1014,239 @@ OO.Ads.manager(function(_, $) {
+     * URLs exist.
+     */
+    var _getNonLinearClickTrackingUrls = function(adObject) {
+      var nonLinearClickTrackingUrls = null;
+      if (adObject &&
+          adObject.ad &&
+          adObject.ad.data &&
+          adObject.ad.data.nonLinear &&
+          adObject.ad.data.nonLinear.nonLinearClickTracking &&
+          adObject.ad.data.nonLinear.nonLinearClickTracking.length > 0) {
+        nonLinearClickTrackingUrls = adObject.ad.data.nonLinear.nonLinearClickTracking;
+      }
+      return nonLinearClickTrackingUrls;
+    };
+
+    /**
+     * Helper function to get an ad object's "high level" click through url.
+     * @private
+     * @method Vast#_getHighLevelClickThroughUrl
+     * @param {object} adObject The ad metadata
+     * @returns {string|null} The clickthrough URL string. Returns null if one does not exist.
+     */
+    var _getHighLevelClickThroughUrl = function(adObject) {
+      var highLevelClickThroughUrl = null;
+      if (adObject &&
+          adObject.ad &&
+          adObject.ad.data &&
+          adObject.ad.data.clickThrough) {
+        highLevelClickThroughUrl = adObject.ad.data.clickThrough;
+      }
+      return highLevelClickThroughUrl;
+    };
+
+    /**
+     * Helper function to get an ad object's "ooyala" click through url.
+     * @private
+     * @method Vast#_getOoyalaClickThroughUrl
+     * @param {object} adObject The ad metadata
+     * @returns {string|null} The clickthrough URL string. Returns null if one does not exist.
+     */
+    var _getOoyalaClickThroughUrl = function(adObject) {
+      var ooyalaClickThroughUrl = null;
+      if (adObject &&
+          adObject.click_url) {
+        ooyalaClickThroughUrl = adObject.click_url;
+      }
+      return ooyalaClickThroughUrl;
+    };
+
+    /**
+     * Helper function to retrieve the ad object's tracking urls under a specific event name.
+     * @private
+     * @method Vast#_getTrackingEventUrls
+     * @param {object} adObject The ad metadata
+     * @param {string} trackingEventName The name of the tracking event
+     * @returns {string[]|null} The array of tracking urls associated with the event name. Returns null if no URLs exist.
+     */
+    var _getTrackingEventUrls = function(adObject, trackingEventName) {
+      var trackingUrls = null;
+      if (adObject &&
+          adObject.ad &&
+          adObject.ad.data &&
+          adObject.ad.data.tracking &&
+          adObject.ad.data.tracking[trackingEventName] &&
+          adObject.ad.data.tracking[trackingEventName].length > 0) {
+        trackingUrls = adObject.ad.data.tracking[trackingEventName];
+      }
+      return trackingUrls;
+    };
+
+
+    /**
+     * Helper function to ping URLs in each set of tracking event arrays.
+     * @private
+     * @method Vast#_pingTrackingUrls
+     * @param {object} urlObject An object with the tracking event names and their
+     * associated URL array.
```

Comment:
Can we include a definition of this url object? Like the keys and possible values associated with it.